site/models/QuestionForm.php

```diff
@@ -105,31 +105,39 @@ public function deleteToday() {
     );
   }
 
+  public function getPrefixProps($prefix) {
+    return array_filter(get_object_vars($this), function($v, $k) use($prefix) {
+      if(strpos($k, $prefix) === 0 && strlen($v)) {
+        return true;
+      }
+    }, ARRAY_FILTER_USE_BOTH);
+  }
+
   public function saveAnswers() {
     $result = true;
-    for($i = 2; $i < 8; $i ++) {
-      $option_id = "user_option_id".$i;
-      if(!empty($this->$option_id)) {
-        $user_option = UserOption::find()->with("option")->where(['id' => $this->{"user_option_id".$i}])->one();
-        for($j = 1; $j < 4; $j ++) {
-          $answer_prop = "answer_".$i.Question::$TYPES[$j];
-          if(!empty($this->$answer_prop)) {
-            $user_option_prop = "user_option_id".$i;
-            $model = new Question;
-            $model->user_id = Yii::$app->user->id;
-            $model->option_id = $user_option->option->id;
-            $model->user_option_id = $this->$user_option_prop;
-            $model->date = new Expression("now()::timestamp");
-            $model->question = $j;
-            $model->answer = $this->$answer_prop;
-            if(!$model->save()) {
-              $result = false;
-            }
-          }
+    $user_bhvr_ids = $this->getPrefixProps('user_option_id');
+    $flipped = array_flip($user_bhvr_ids);
+    $user_bhvrs = UserOption::findAll(array_values($user_bhvr_ids));
```

Comment:
This whole function really needs tests but this line (DB call) makes that difficult. Figure out a way to get this unit tested.